### PR TITLE
Tests: Use portable #! in python scripts (/usr/bin/env)

### PR DESCRIPTION
**Pull request by ghost**
These changes should also be backported to the stable releases so that they build and test on systems which don't install python to /usr

It also changes python scripts which are not related to tests because it was easier not to have grep discriminate.

### CODE REVIEW

File: contrib/devtools/security-check.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python2
+#!/usr/bin/env python2
```
Comment (MarcoFalke) [inline contrib/devtools/security-check.py]: Couldn't this say `#!/usr/bin/env python`, since #7723?

Comment (laanwj) [inline contrib/devtools/security-check.py]: Yes, `security-check` works with python 3 too. The build system calls all these scripts by passing them to the python interpreter, so it doesn't care what is on the first line.

Comment (ChoHag) [inline contrib/devtools/security-check.py]: I made no attempt to check for portability of the python scripts between python releases, just a straight s/// on all *.py files

Comment (MarcoFalke) [inline contrib/devtools/security-check.py]: Can you remove the trailing 2's according to the pull request I linked above?

Comment (ChoHag) [inline contrib/devtools/security-check.py]: There are 5 files using python2, only security-check.py and symbol-check.py (notably not test-security-check.py) have been changed in #7723. Accordingly I've changed only the same two files.

contrib/devtools/security-check.py:#!/usr/bin/env python2
contrib/devtools/symbol-check.py:#!/usr/bin/env python2
contrib/devtools/test-security-check.py:#!/usr/bin/env python2
contrib/zmq/zmq_sub.py:#!/usr/bin/env python2
share/rpcuser/rpcuser.py:#!/usr/bin/env python2 

I'm happy to abandon this PR in favour of incorporating the changes directly in #7723 if it simplifies things.

File: contrib/devtools/security-check.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python2
```
Comment (petertodd) [inline contrib/devtools/security-check.py]: Did we actually test that this works on python3?

Comment (fanquake) [inline contrib/devtools/security-check.py]: @petertodd See @laanwj's comment https://github.com/bitcoin/bitcoin/pull/8270#discussion_r68552510

**Comment by petertodd** [comment]
utACK 7b01ce254cfa0e8ae7e57a72c57e8ae89f3c1829

**Comment by fanquake** [comment]
utACK 7b01ce2

**Comment by MarcoFalke** [comment]
utACK 7b01ce2

**Pull request merged**